### ISSUE TITLE
Swift: Update Docs/Example Repos with use of dynamic keyword

### ISSUE THREAD

**Issue by natevecc** [cloudmine/cloudmine-ios#46]
Was running into an issue where objects weren't updating and hunted it down to the fact that Swift doesn't support KVO out of the box: [docs](https://developer.apple.com/library/mac/documentation/Swift/Conceptual/BuildingCocoaApps/AdoptingCocoaDesignPatterns.html#//apple_ref/doc/uid/TP40014216-CH7-XID_8)(see section `Key-Value Observing`).

Long story short when using Swift with the cloudmine-ios library make sure to add dynamic in objects that extend CMObject:
```swift
class MyClass: CMObject {
  // the right way. changes to this property will be picked up by KVO, the object will be marked 
  // dirty and saved when calling #save() or #saveWithUser()
  dynamic var myProperty: String?
  
  // the wrong way. changes will not be picked up by KVO and the object will never be marked 
  // as dirty and saved correctly
  var anotherProperty: String?
}
```

**Comment by ethanmick** [comment]
@pbenmoser4 @jraymondcm

**Comment by jraymondcm** [comment]
@ethanmick thanks. We'll add a note to the documentation. 

Do we know if extending CMObject is the only circumstance under which this would be necessary?

**Comment by natevecc** [comment]
Also found another work around that works well with enums which don't support the `dynamic` keyword.
```swift
enum State: String {
  case Phase1
  case Phase2
}

class MyClass: CMObject {
  var myProperty: State {
    didSet: {
      self.dirty = true
    }
  }
}
```

**Comment by ethanmick** [comment]
@jraymondcm Yes, since CMObjects are the only class that uses the KVO to set themselves to dirty.

@natevecc Thanks for that example too. We should mention the CMObject dirty setting in our docs. I'll talk to @jraymondcm
